Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -16,32 +17,94 @@
 #include <QKeySequence>
 #include <QLabel>
 #include <QLineEdit>
+#include <QListWidget>
 #include <QMessageBox>
 #include <QPushButton>
 #include <QScrollArea>
 #include <QString>
 #include <QStringList>
+#include <QThread>
 #include <QVBoxLayout>
 
 #include "Common/IOFile.h"
 
+#include "Common/FileUtil.h"
 #include "Core/Config/MainSettings.h"
+#include "Core/IOS/USB/Emulated/Skylander.h"
 #include "Core/System.h"
+#include "DolphinQT/RenderWidget.h"
+#include "DolphinQt/MainWindow.h"
 
 #include "DolphinQt/QtUtils/DolphinFileDialog.h"
 #include "DolphinQt/Resources.h"
 #include "DolphinQt/Settings.h"
 
-// Qt is not guaranteed to keep track of file paths using native file pickers, so we use this
-// static variable to ensure we open at the most recent Skylander file location
-static QString s_last_skylander_path;
+// PortalButton
+PortalButton::PortalButton(RenderWidget* rend, QWidget* pWindow, QWidget* parent) : QWidget(parent)
+{
+  SetRender(rend);
+  portal_window = pWindow;
+
+  setWindowTitle(tr("Portal Button"));
+  setWindowFlags(Qt::Widget | Qt::FramelessWindowHint);
+  setParent(0);
+  setAttribute(Qt::WA_NoSystemBackground, true);
+  setAttribute(Qt::WA_TranslucentBackground, true);
+
+  button = new QPushButton(tr("Portal of Power"), this);
+  button->resize(100, 50);
+  connect(button, &QAbstractButton::clicked, this, [this]() { OpenMenu(); });
+  fade_out.callOnTimeout(this, &PortalButton::hide);
+
+  move(100, 150);
```

Comment:
When in fullscreen (on mac) this continuously toggles between 2 separate fullscreen components. Granted, this is not the majority of the user base, but if we give them the option then it will need to work as expected